### PR TITLE
Fixed NoData flicker while loading repo list

### PR DESCRIPTION
**Pull request by vivek-harness**
Fix for https://harness.atlassian.net/browse/PIPE-23625

Before the fix, a NoData UI appeared when the repo list page was refreshed. This PR fixes the issue.

https://github.com/user-attachments/assets/f22c1e71-e887-46a6-954c-f257379a0e25

### CODE REVIEW

File: packages/ui/src/components/loaders/skeleton-list.tsx
Comment (knagurski) [inline packages/ui/src/components/loaders/skeleton-list.tsx:1]: I think this exists outside of the `loaders` directory. Can we drop this and use that?

**Review by praneshg239 — APPROVED** [review]

**Review by knagurski — CHANGES_REQUESTED** [review]
Looks like extra stuff snuck in

**Comment by praneshg239** [comment]
@vivek-harness all the lint rules are failing. Please check.

File: packages/ui/src/components/loaders/skeleton-list.tsx
Comment (vivek-harness) [reply packages/ui/src/components/loaders/skeleton-list.tsx:1]: Ohh, I see that it got added as Skeleton. It wasn't there when I started working on this. Anyway, I just deleted the skeleton-list component that I moved over from canary to ui, and used Skeleton instead.

File: packages/ui/src/components/index.ts [hunk 1]
```diff
@@ -31,10 +31,8 @@ export * from './commit-copy-actions'
 export * from './textarea'
 export * from './accordion'
 export * from './breadcrumb'
-export * from './skeleton-list'
```
Comment (knagurski) [inline packages/ui/src/components/index.ts:34]: We'll still need this line too

File: packages/ui/src/components/skeleton.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { cn } from '@utils/cn'
 
 function Skeleton({ className, ...props }: React.HTMLAttributes<HTMLDivElement>) {
-  return <div className={cn('bg-background-3 animate-pulse rounded-3xl', className)} {...props} />
```
[the remark this answers is not in the document]
Comment (vivek-harness) [reply packages/ui/src/components/skeleton.tsx]: Yup, I got a few merge conflicts. I fixed them.

File: packages/ui/src/components/index.ts (hunk 1 above)
Comment (vivek-harness) [reply packages/ui/src/components/index.ts]: Yup, I fixed it.

**Pull request merged**